x-pack/legacy/plugins/code/server/lsp/abstract_launcher.ts

```diff
@@ -15,17 +15,21 @@ import { RequestExpander } from './request_expander';
 import { ControlledProgram } from './process/controlled_program';
 
 let seqNo = 1;
+// Wait this time before try to reconnect.
+// Sometimes a sub process may dies or disconnected before kibana receives a sigint
+// We use this time to prevent restart a new process during the shutdown process.
+const RECONNECT_INTERVAL = 5000;
 
 export abstract class AbstractLauncher implements ILanguageServerLauncher {
   running: boolean = false;
   private currentPid: number = -1;
   private child: ControlledProgram | null = null;
   private startTime: number = -1;
-  private proxyConnected: boolean = false;
```

Comment:
Just curious, this line delete has no relationship with the relaunch too quick issue, right?